Create DRCF digital markets research finder

baisa opened the pull request:
Following CMA request to create a specialist finder to bring together research from several organisations.

[Trello card](https://trello.com/c/0LHrDag7/157-create-new-specialist-finder-for-competition-and-markets-authority)

baisa committed:
Create DRCF digital markets research finder

Following CMA request to create a specialist finder to bring together research from several organisations.

sihugh commented on lib/documents/schemas/drcf_digital_markets_researches.json:
```diff
@@ -0,0 +1,115 @@
+  "name": "DRCF digital markets research",
+  "description": "DRCF digital markets research finder",
+  "filter": {
+    "format": "find_digital_market_research"
```
This should be `drcf_digital_markets_research` to match the content type

baisa committed:
Add validation according to requirements

Also added test coverage to ensure the process of creating DRCF digital markets research is working correctly.

kevindew commented on app/views/metadata_fields/_drcf_digital_markets_researches.html.erb:
```diff
@@ -0,0 +1,61 @@
+          placeholder: 'Select a category'
+        }
+      }
+  %>
```
As I understand we were changing this to be a single select rather than a select 2?

baisa replied:
I have added `include_blank: true`, as single select would break the specs, but maybe I am missing something.

kevindew replied:
Sorry, this was my mistake. I thought it still remained a select multiple because it was a select2. I now notice there isn't the `multiple: true` attribute.

Is the precedence in this app to always use a select2 for any select inputs? It seems a bit heavy duty to me given there's only 2 options here and a non-enhanced select would be more appropriate.

baisa replied:
Yes it looks like specialist-publisher uses select2 exclusively.

kevindew replied:
Cool - lets keep that then. Sorry for the misunderstanding

kevindew commented on spec/fixtures/factories.rb:
```diff
@@ -645,4 +650,20 @@
+          "digital_market_research_category" => "ad-hoc-research",
+          "digital_market_research_publisher" => "gambling-commission",
+          "digital_market_research_area" => "media-and-entertainment",
+          "digital_market_research_topic" => "future-connectivity",
```
I imagine that these should be arrays since multiple can be selected.

baisa replied:
Quite right! 👍

kevindew commented on spec/features/creating_a_drcf_digital_markets_research_spec.rb:
```diff
@@ -0,0 +1,153 @@
+    expect(page).to have_content("Please fix the following errors")
+    expect(page).to have_content("Title can't be blank")
+    expect(page).to have_content("Summary can't be blank")
+    expect(page).to have_content("Body can't be blank")
```
Is this missing a category validation?

baisa replied:
Yes, I have added it, thanks!

kevindew commented on spec/features/creating_a_drcf_digital_markets_research_spec.rb:
```diff
@@ -0,0 +1,153 @@
+
+    expect(page).to have_css(".elements-error-summary")
+    expect(page).to have_css(".form-group.elements-error")
+    expect(page).to have_css(".elements-error-message")
```
I can see this is in some of the other tests but not too sure of the value of it, they just tell us some html elements exists which isn't particularly of value to a user - would suggest removing these lines.

kevindew reviewed (commented):
The tests look satisfyingly consistent with the other ones in this repo - nice job.

baisa committed:
Hardcode primary_publishing_organisation as CMA
